tags/logrus/server_interceptors.go

```diff
@@ -0,0 +1,32 @@
+package ctxlogger_logrus
+
+import (
+	"github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// PayloadUnaryServerInterceptor returns a new unary server interceptors that adds logrus.Entry to the context.
+func UnaryServerInterceptor(entry *logrus.Entry) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		newCtx := newLoggerForCall(ctx, entry)
+		resp, err := handler(newCtx, req)
+		return resp, err
+	}
+}
+
+// StreamServerInterceptor returns a new streaming server interceptor that adds logrus.Entry to the context.
+func StreamServerInterceptor(entry *logrus.Entry) grpc.StreamServerInterceptor {
+	return func(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+		newCtx := newLoggerForCall(stream.Context(), entry)
+		wrapped := grpc_middleware.WrapServerStream(stream)
+		wrapped.WrappedContext = newCtx
+		err := handler(srv, wrapped)
+		return err
+	}
+}
```

Comment:
These methods should probably stay in `logging/`. I think ctxlogger package shouldn't have any grpc specific code. Happy to discuss.

Reply:
Yeah, I've kind of been thinking about this one as well ... been going over it in my head abit trying to weigh up both approaches.